doc/reference/src/low-level-execution-model.pdc

```diff
@@ -111,276 +111,542 @@ The oracle-free abstract algorithm for optimal reduction operates on four node t
+          \outwire[]{b/1}{c}
+          \outwire[]{b/4}{b}
+          \cutWire{a.out}{b.out}{}{}
+        \end{tikzpicture}
+      \end{subfigure}
+      ~
+      \begin{subfigure}[c]{0.05\textwidth}
+        \huge{→}
+      \end{subfigure}
+      ~
+      \begin{subfigure}[c]{0.1\textwidth}
+        \begin{tikzpicture}
+          \inetcell[rotate = 180]{a}{\rotatebox[origin=c]{180}{$a$}}[5]
+          \inetcell[rotate = 180, right = 60 pt of a]{b}{\rotatebox[origin=c]{180}{$a$}}[5]
+          \inetcell[below = 40 pt of a]{c}{$f_i$}[5]
+          \inetcell[below = 40 pt of b]{d}{$f_i$}[5]
+          \inwire[]{a.out}{a}
+          \inwire[]{b.out}{d}
+          \outwire[]{c.out}{b}
+          \outwire[]{d.out}{c}
+          \swire[0]{a/4}{c/1}{}{}
+          \swire[0]{b/1}{d/4}{}{}
+          \swire[0]{a/1}{d/1}{}{}
+          \swire[0]{b/4}{c/4}{}{}
+        \end{tikzpicture}
+      \end{subfigure}
+    \end{figure}
+    Here we see the case for a fan-in node $f_i$ and some arbitrary node $a$. Both nodes get duplicated,
+    however the number of free ports is preserved, as each node that is created is internally connected to the duplicated nodes.
+    Additionally the four external ports are preserved among the four duplicated nodes.
+  \end{tcolorbox}
+  $\therefore$ since all cases are covered, $Eval(A)$ has only one free port.
+\end{Theorem}
+
+We now shall define one more definition before getting to our main theorem.
+
+
+\begin{Definition}{Valid EAL-Net}{Valid EAL-Net}
+  A \textcolor{definitions}{\textbf{Valid EAL-Net}} is a net translated from the EAL subset subset of BOHM and has undergone zero or more reduction steps.
+\end{Definition}
+
+\begin{Theorem}{Read back from a Valid EAL-Net gives is $α$ equivalent to the normal evaluation of the original EAL-term}{readback}
+  Let $A$ be the Valid EAL-Net.
+  \\ \\
+  By Definition \ref{De:Valid EAL-Net} this net originated from a valid EAL-AST which is a subset of BOHM.
+  \\ \\
+  Thus by Theorem \ref{Th:AST->Net->Eval} $A$ must only have one free port.
+  \\ \\
+  Furthermore this free port must be the root of the graph/old AST.
+  \\ \\
+  Now to show this evaluates to the same answer, we must reconstruct the ADT considering all cases.
+  \\ \\
+  For this we must keep two maps as we do this algorithm.
+  \\ \\
+  The first being the variable name from a specific lambda.
+  \\ \\
+  The second map is a map from the fan in number to the current status of our traversal of said fan-in.
+  \\ \\
+  We can now a proof by induction, specifically, we shall consider three cases:
+
+  1. $\lambda$ /\ $\mu$
+
+  2. $FanIn$
+
+  3. The rest.
+
+  We will also consider the inductive case first for the following proof.
+  \\ \\
+  Let $rec$ be this recursive algorithm.
+  \begin{tcolorbox}[title= Case 1 - rest, colframe=inner-box-1]
+    The logic for all other cases is simple. Consider some node $P$ with ports $1 \cdots n$.
+    \\ \\
+    Let port n be the parent ADT port.
```

Comment:
AST?

Reply:
AST/ADT

Reply:
I don't think this usage is standard. A BOHM term isn't an ADT - an ADT is the *type*.

Maybe a BOHM term is "an inhabitant of the BOHM ADT", if you like.